stdlib/public/Concurrency/AsyncSequence.swift

```diff
@@ -113,6 +285,26 @@ func _first<Source: AsyncSequence>(
 
 @available(SwiftStdlib 5.5, *)
 extension AsyncSequence {
+  /// Returns the first element of the sequence that satisfies the given
+  /// predicate.
+  ///
+  /// In this example, an asynchronous sequence called `Counter` produces `Int`
+  /// values from `1` to `10`. The `first(where:)` method returns the first
+  /// member of the sequence that's evenly divisible by both `2` and `3`.
+  ///
+  ///     let divisibleBy2And3 = await Counter(howHigh: 10)
+  ///         .first (where: { $0 % 2 == 0 && $0 % 3 == 0 } )
```

Comment:
does this match the documentation style for `Sequence`? e.g. should it be a trailing closure?

Also there is a rogue space after `first` and before the `(`

Reply:
`Sequence.first(where:)` in the stdlib doesn't use a trailing closure for its docs: https://developer.apple.com/documentation/swift/sequence/2907979-first

```
let numbers = [3, 7, 4, -2, 9, -6, 10, 1]
if let firstNegative = numbers.first(where: { $0 < 0 }) {
    print("The first negative number is \(firstNegative).")
}
// Prints "The first negative number is -2."
```

With such a short closure, leaving it there next to the `where:` parameter may improve readability for newcomers.

Reply:
Update: found a clean way to use trailing closures, did that everywhere in f63ca1d.